convert.py

```diff
@@ -366,44 +366,47 @@ def __init__(self, fname_tokenizer: Path, fname_added_tokens: Path | None) -> No
             added_tokens = {}
 
         vocab_size: int = self.sentencepiece_tokenizer.vocab_size()
-        expected_ids = list(range(vocab_size, vocab_size + len(added_tokens)))
-        actual_ids   = sorted(added_tokens.values())
-        if expected_ids != actual_ids:
-            raise Exception(f"Expected added token IDs to be sequential and start at {len(added_tokens)}; got {actual_ids}")
 
-        items = sorted(added_tokens.items(), key=lambda text_idx: text_idx[1])
-        self.added_tokens_list = [text for (text, idx) in items]
-        self.vocab_size_base: int = vocab_size
-        self.vocab_size: int = self.vocab_size_base + len(self.added_tokens_list)
-        self.fname_tokenizer = fname_tokenizer
-        self.fname_added_tokens = fname_added_tokens
+        new_tokens: dict[int, str] = {id: piece for piece, id in added_tokens.items() if id >= vocab_size}
+        expected_new_ids: list[int] = list(range(vocab_size, vocab_size + len(new_tokens)))
+        actual_new_ids: list[int]   = sorted(new_tokens.keys())
```

Comment:
Are these type annotations actually necessary? Usually type checkers like mypy can infer the type of a variable based on the value that is assigned to it.